backend/src/utils/constants.ts

```diff
@@ -130,3 +130,6 @@ export const DEFAULT_NOTEBOOK_SIZES: NotebookSize[] = [
     },
   },
 ];
+
+export const imageUrlRegex =
+  /^([\w.\-_]+((?::\d+|)(?=\/[a-z0-9._-]+\/[a-z0-9._-]+))|)(?:\/|)([a-z0-9.\-_]+(?:\/[a-z0-9.\-_]+|))(?::([\w.\-_]{1,127})|)/;
```

Comment:
This regex won't match `@sha256...` as the tag because if use that, we cannot set it as the tag name. So in the custom image, the tag name will be set to `latest` but the repo URL will be set to the correct one with `@sha256`.

Reply:
Oh my gosh -- we need you to comment this in some fashion haha... wowzers, that's a regex and a half haha.